File: result/metagpt/claude/humaneval/22.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Define a structure to hold any type of value
typedef struct {
    enum { INT, DOUBLE, STRING, OTHER } type;
    union {
        int int_value;
        double double_value;
        char* string_value;
        void* other_value;
    };
} any;

// Define a structure to hold a list of any values
typedef struct {
    any* items;
    size_t size;
    size_t capacity;
} list_any;

// Function to filter integers from a list of any values
int* filter_integers(list_any values, size_t* out_size) {
    int* out = NULL;
    size_t count = 0;

    for (size_t i = 0; i < values.size; i++) {
        if (values.items[i].type == INT) {
            out = realloc(out, (count + 1) * sizeof(int));
            out[count] = values.items[i].int_value;
            count++;
        }
    }

    *out_size = count;
    return out;
}

File: result/metagpt/gpt4o/humaneval/22.rs
enum Any {
    Int(i32),
    Double(f64),
    String(String),
    Other(Box<dyn std::any::Any>),
}

struct ListAny {
    items: Vec<Any>,
}

fn filter_integers(values: &ListAny) -> Vec<i32> {
    let mut out = Vec::new();

    for item in &values.items {
        if let Any::Int(value) = item {
            out.push(*value);
        }
    }

    out
}

fn main() {
    let values = ListAny {
        items: vec![
            Any::Int(42),
            Any::Double(3.14),
            Any::String(String::from("hello")),
            Any::Int(7),
        ],
    };

    let integers = filter_integers(&values);
    println!("Filtered integers: {:?}", integers);
}